app/domain/authentication/authn_azure/application_identity.rb

```diff
@@ -2,30 +2,43 @@ module Authentication
   module AuthnAzure
 
     Log = LogMessages::Authentication::AuthnAzure
+    Err = Errors::Authentication::AuthnAzure
 
     class ApplicationIdentity
 
-      def initialize(role_annotations:, service_id:, logger:)
+      attr_reader :resources, :resource_types
```

Comment:
IMO helpful to add some background and explanatory comments here

Reply:
would this help?
```
      # resources: list of AzureResource objects included in the Application Identity
      # resource_types: list of the Azure resource types included in the Application Identity
      attr_reader :resources, :resource_types
```

If not, what data do you think is missing?

Reply:
I feel like resources and resource_type are too similiar and took me a while to understand even with your comment which leads me to believe naming will help solve the problem.

resources is just a list of AzureResources that comes from the application identity of the host.
so what about `ApplicationIdentitiesObject` or `AzureApplicationIdentitiesList` or `HostApplicationIdentitiesList`... Whatever you choose it should be clear that this object holds a list of Host application identities 

As mentioned in a comment below, I think it would be more representative if `resource_types` would be `resource_name` or even `application_identity_name`

Reply:
I would rather not to add Application Identity to the name as i want to rename `ApplicationIdentity` to something else. I thought of doing this in another PR to make this PR smaller but maybe it makes sense to have them in the same PR. 

what do you think about `ResourceRestrictions`? This name is more self-explanatory as this is what the application identity is. It collects restrictions of resources from which the host can authenticate.

Reply:
I am interested to hear your thought process around ResourceRestrictions...  I remember us talking in the scope of Azure about how this collection (subscription id, resource group, etc) is the native identity of the resource so wouldn't at least `resource_identity` be fitting?

I understood that subscription id, resource group, etc are affiliations. They are define the azure identity of a VM, not what it restricts it from authenticating but again, let's talk about this more

Reply:
well, the word "identity" is not in place as these objects are not part of any identity. Is it part of the Conjur identity? Is it part of the Azure identity? The answer to both questions is "no". It is not the id of either.

Then what are these characteristics? If you explain them to me in words, you will say something like "they define objects in the native environment from which the host can authenticate with Conjur", right? For example, if the host has an annotation `authn-azure/resource-group: some-rg` then it means that this host can authenticate with the Azure authenticator only with an Azure AD token that was enrolled for an Azure resource in the `some-rg` resource group. So it is basically a restriction on the resource.

Another thing that led me to this is that when I tried to explain what an Application Identity is, I gave the analogy of IP restrictions. Where if a host has the field `restricted_to: 1.1.1.1` then it means that it can authenticate with Conjur only from the IP 1.1.1.1. The language is pretty much the same. And that led me to understand that application identities are basically restrictions on the resource that the host is authenticating from, and not an identity at all.

Reply:
Is `some-rg` not part of it's identity? The resource is _**part**_ of a grouping called `some-rg` which is why "affiliation" is more descriptive . Or another example, for the system assigned ID, the OID of the resource, it makes up its identity no? 

It seems to me that "affiliation" would be more fitting. What about `ResourceAffiliation`?

Reply:
Why is it better than Restriction? I don't like words that send me to google translate :)

Restriction is _exactly_ what the thing is.

Reply:
we'll wait for @jonahx to give his opinion. I think that he has the value of someone who didn't implement the Application Identity and has a fresh view on it.

Reply:
I agree `Restriction` feels like an accurate, good choice here.

Reply:
> I understood that subscription id, resource group, etc are affiliations. They are define the azure identity of a VM, not what it restricts it from authenticating but again, let's talk about this more

To answer @sigalsax 's question directly, I think in another context the affiliations perspective might be appropriate, as might an "attribute" perspective.  But in the context of an authenticator, we're saying, "if you, the application, don't check of all these boxes, you're not allowed to authenticate".  Hence "restriction" feel accurate.  I think "requirements" could work for the same reason.

Reply:
Thanks, I will change it to `ResourceRestriction` and we can continue from there.